In [2]:
import pandas as pd
import plotly.express as px

df  = pd.read_csv('dataset_2.csv')


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 119 entries, 0 to 118
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             119 non-null    object 
 1   Trips Completed  119 non-null    int64  
 2   Accept Rate      119 non-null    object 
 3   Supply Hours     119 non-null    int64  
 4   Rating           119 non-null    float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.8+ KB


In [4]:
df.head()

Name  Trips Completed Accept Rate  Supply Hours  Rating
0    Abdul                1        100%             3     4.8
1  Abraham               12         83%             5     4.7
2  Adelina                1        100%             2     4.7
3   Akilah                1        100%             2     4.9
4     Alec               21         76%            11     5.0

In [5]:
df.isnull().sum()

Name               0
Trips Completed    0
Accept Rate        0
Supply Hours       0
Rating             0
dtype: int64

In [6]:

df['Accept Rate'] = df['Accept Rate'].str.replace('\D+', '')
df['Accept Rate'] = pd.to_numeric(df['Accept Rate'])


C:\Users\hs111\AppData\Local\Temp/ipykernel_17220/3021448889.py:1: FutureWarning: The default value of regex will change from True to False in a future version.
  df['Accept Rate'] = df['Accept Rate'].str.replace('\D+', '')


In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 119 entries, 0 to 118
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             119 non-null    object 
 1   Trips Completed  119 non-null    int64  
 2   Accept Rate      119 non-null    int64  
 3   Supply Hours     119 non-null    int64  
 4   Rating           119 non-null    float64
dtypes: float64(1), int64(3), object(1)
memory usage: 4.8+ KB


In [8]:
df[df.duplicated(subset='Name')]


Empty DataFrame
Columns: [Name, Trips Completed, Accept Rate, Supply Hours, Rating]
Index: []

# Total bonus payout 
## Option 1

In [20]:
option1drivers = df[(df['Supply Hours']>=8)&(df['Accept Rate']>90)&(df['Rating']>=4.7)&(df['Trips Completed']>=10)]
len(option1drivers)*50


950

the bonus amount is going to be 950 dollars in total. if the saturday is going to be as busy as it was in the data. 

## Option 2

In [25]:
option2drivers = df[(df['Trips Completed'] >= 12)&(df['Rating']>=4.7)]
sum(option2drivers['Trips Completed'])*4


2976

The bonus amount under option 2 is going to be higher than the option 1 by about 30%.

## Option 1 but not Option 2

In [27]:

bothOption = option1drivers.merge(option2drivers ,on = 'Name', how = "inner")
justUnderOption1 = len(option1drivers) - len(bothOption)
justUnderOption1


1

Only 1 driver would qualify for option 1 but not under option 2

In [36]:
Question4dataset = df[(df['Trips Completed']<10)&(df['Accept Rate']<90)&(df['Rating']>=4.7)]
print(round((len(Question4dataset)/len(df))*100,2),'%')


10.92 %


10.92 % percentages of drivers online completed less than 10 trips, had an acceptance rate of less than 90%, and had a rating of 4.7 or higher

# Scenario 2

## Rent Vehicle expenses Taxi Driver

In [38]:
(6*200) - 200 - 400 - (500)

100

## Own car and partner with Uber

In [39]:
(6*200) - 200*1.05 - 400*0.8 
# still need to buy a car , car cost not included

670.0

### Assuming the driver only takes 3 weeks off in both the scenarios the calcualtion would seem something like this

### there are 52.14 weeks in a year 

In [40]:
(49.4*6*200) - (49.4*200) - (12*400) - (49.4*500)

19899.999999999993

### Taxi driver makes about 19900 dollars in a year without partenering with uber

In [41]:
(49.4*200*1.05) + (12*400*0.8) + (40000) +(19900)


74114.0

In [42]:
49.4*6


296.4

In [43]:
74114/296

250.38513513513513

### 250 instead of 200 , they need to increase by about 25  percent